$OpenBSD: patch-wscript,v 1.1.1.1 2010/12/28 18:19:28 jasper Exp $

- OpenBSD can also use libkvm and libexecinfo.
- The SSL tests fail as -lssl isn't appended to LDFLAGS, override it.

--- wscript.orig	Wed Nov 17 06:46:59 2010
+++ wscript	Fri Dec 10 16:38:54 2010
@@ -155,7 +155,7 @@ def configure(conf):
     conf.env.append_value("CCFLAGS", "-rdynamic")
     conf.env.append_value("LINKFLAGS_DL", "-rdynamic")
 
-  if sys.platform.startswith("freebsd"):
+  if sys.platform.startswith("freebsd") or sys.platform.startswith("openbsd"):
     conf.check(lib='kvm', uselib_store='KVM')
 
   #if Options.options.debug:
@@ -164,7 +164,7 @@ def configure(conf):
   if Options.options.efence:
     conf.check(lib='efence', libpath=['/usr/lib', '/usr/local/lib'], uselib_store='EFENCE')
 
-  if sys.platform.startswith("freebsd"):
+  if sys.platform.startswith("freebsd") or sys.platform.startswith("openbsd"):
      if not conf.check(lib="execinfo",
                        includes=['/usr/include', '/usr/local/include'],
                        libpath=['/usr/lib', '/usr/local/lib'],
@@ -321,6 +321,7 @@ def configure(conf):
   default_compile_flags = ['-g', '-O3']
   conf.env.append_value('CCFLAGS', default_compile_flags)
   conf.env.append_value('CXXFLAGS', default_compile_flags)
+  conf.env.append_value('${CFLAGS}', default_compile_flags)
   conf.write_config_header("config.h")
 
 
@@ -573,11 +574,10 @@ def build(bld):
   # Only install the man page if it exists.
   # Do 'make doc install' to build and install it.
   if os.path.exists('doc/node.1'):
-    bld.install_files('${PREFIX}/share/man/man1/', 'doc/node.1')
+    bld.install_files('${PREFIX}/man/man1/', 'doc/node.1')
 
   bld.install_files('${PREFIX}/bin/', 'bin/*', chmod=0755)
-  bld.install_files('${PREFIX}/lib/node/wafadmin', 'tools/wafadmin/*.py')
-  bld.install_files('${PREFIX}/lib/node/wafadmin/Tools', 'tools/wafadmin/Tools/*.py')
+  bld.install_files('${PREFIX}/lib/node/wafadmin', 'tools/wafadmin/Node.py')
 
 def shutdown():
   Options.options.debug
